File: 2023/Leetcode/problem136.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        numsList = []

        for i in range(0, len(nums)):
            if nums[i] not in numsList:
                numsList.append(nums[i])

            else:
                numsList.pop(numsList.index(nums[i]))

        return numsList[0]

File: 2023/Leetcode/problem136.cs
public class Solution {
    // n = Length of nums
    // Time Complexity: n = O(n) Linear Time
    // Space Complexity: n / 2 = O(n) Linear Space
    public int SingleNumber(int[] nums) {
        var numsList = new HashSet<int>();

        for (int i = 0; i < nums.Length; i++) // O(n)
        {
            if (!(numsList.Contains(nums[i])))  // O(1)
            {
                numsList.Add(nums[i]); // O(1)
            }
            else
            {
                numsList.Remove(nums[i]); // O(1)
            }
        }

        foreach (int num in numsList)
        {
            return num;
        }
        
        return 0;
    }
}